docs/partitioning.md

```diff
@@ -92,7 +92,7 @@ This leads to several problems:
 
 Iceberg produces partition values by taking a column value and optionally transforming it. Iceberg is responsible for converting `event_time` into `event_date`, and keeps track of the relationship.
 
-Table partitioning is configured using these relationships. The `logs` table would be partitioned by `date(event_time)` and `level`.
+Table partitioning is configured using these relationships. The `logs` table would be partitioned by `date(event_time)` and `level`.  Since the `day` partition transform returns the number of days from 1970-01-01, this table will be partitioned by calendar day and not day of the month.
```

Comment:
I think maybe better to link the iceberg spec here https://iceberg.apache.org/spec/#partition-transforms, it's pretty clear if one want to know more about all types of partitions transformation

<img width="995" alt="image" src="https://github.com/apache/iceberg/assets/5961173/51350c3f-93f3-4b2b-ab65-e352f47d0945">

Reply:
@dramaticlly -- I updated the PR and changed only the partition transform in the example.  The link you suggested is already on the page just below the section I changed.